apps/studio/src/server/modules/page/page.router.ts

```diff
@@ -40,26 +42,32 @@ export const pageRouter = router({
   updatePage: protectedProcedure
     .input(updatePageSchema)
     .query(async ({ input, ctx }) => {
-      const parentId = ''
-      const pageName = ''
+      // NOTE: Need to do validation like checking for existence of the page
+      // and whether the user has write-access to said page
+      await updatePageById(input)
 
       return {
         // NOTE: This should adhere to our db schema
-        parentId,
-        pageName,
+        ...input,
         // NOTE: Don't need `variant` here
         // because the router itself discriminates for us
       }
     }),
+
   updatePageBlob: protectedProcedure
     .input(updatePageBlobSchema)
     .query(async ({ input, ctx }) => {
+      // NOTE: Need to do validation like checking for existence of the page
+      // and whether the user has write-access to said page
+      await updateBlobById(input)
+
       // NOTE: Not returning the `content` first because
       // 1. it might potentially be huge
       // 2. frontend already knows
       // ahead of time what is the content (it sent the content over)
```

Comment:
am ok with returning the input (might be huge, but i think for one agency dont have to think too much about perf for now) 
just remove this no longer relevant comment can already

Reply:
shure